File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int get_change(int);
float myInput;

int main(void)
{
    // continue prompting user until a float above 0 is entered
    do
    {
        myInput = get_float("Change owed: ");
    }
    while (myInput < 0);
    // convert float to integer for math operations
    int cents = round(myInput * 100);
    int coinCount = get_change(cents);
    printf("%i\n", coinCount);
}

int get_change(int cents)
{
    int coinCount = 0;
    while (cents > 0)
    {
        if (cents >= 25)
        {
            coinCount++;
            cents -= 25;
        }
        else if (cents >= 10)
        {
            coinCount++;
            cents -= 10;
        }
        else if (cents >= 5)
        {
            coinCount++;
            cents -= 5;
        }
        else if (cents >= 1)
        {
            coinCount++;
            cents -= 1;
        }
    }
    return coinCount;
}

File: pset6/cash/cash.py
while True:
    myInput = input("Change Owed: ")
    try:
        if "." in myInput:
            myInput = float(myInput)
            if myInput > 0:
                break
        myInput = int(myInput)
        if myInput > 0:
            break
    except:
        continue
cents = int(myInput * 100)


def getChange(cents):
    coinCount = 0
    while cents > 0:
        if cents >= 25:
            coinCount = coinCount + 1
            cents = cents - 25
        elif cents >= 10:
            coinCount = coinCount + 1
            cents = cents - 10
        elif cents >= 5:
            coinCount = coinCount + 1
            cents = cents - 5
        elif cents >= 1:
            coinCount = coinCount + 1
            cents = cents - 1
    return coinCount


coinCount = getChange(cents)
print(coinCount)